regcore/search/models.py

```diff
@@ -1,11 +1,41 @@
+from datetime import date
+
 from django.db import models
 
 from django.contrib.postgres.fields import ArrayField
-from django.contrib.postgres.search import SearchVector, SearchVectorField
+from django.contrib.postgres.search import (
+    SearchVector,
+    SearchVectorField,
+    SearchQuery,
+    SearchRank,
+    SearchHeadline,
+)
 
 from regcore.models import Part
 
 
+class SearchIndexQuerySet(models.QuerySet):
+    def search(self, query):
+        return self\
+            .filter(part__in=models.Subquery(Part.objects.effective(date.today()).values("id")))\
+            .filter(search_vector=SearchQuery(query))\
+            .annotate(rank=SearchRank("search_vector", SearchQuery(query)))\
+            .annotate(
+                headline=SearchHeadline(
+                    "content",
+                    SearchQuery(query),
+                    start_sel='<span class="search-highlight">',
+                    stop_sel='</span>',
+                ),
+            )\
+            .order_by('-rank')\
+            .values("type", "content", "headline", "label", "parent", "part__document__title", "part__title", "part__date")
```

Comment:
Can get rid of the `values` call from this, it is useful specifically for the API endpoint and should be clear in it's use.